packages/http/libraries/core/src/http/guard/AuthGuard.ts

```diff
@@ -0,0 +1,41 @@
+import { injectable } from 'inversify';
+
+import { UserRequest } from '../models/UserRequest';
+import { Guard } from './Guard';
+
+@injectable()
+export abstract class AuthGuard<
```

Comment:
Let's avoid implementing authentication in this PR. Auth is a complex topic to tackle. I would love to have an abstraction that covers user and non user flows with different flows such as oauth and oidc.

Reply:
Okay, imho we should discuss this more thoroughly. By removing the auth, we are not complying with the current API.

Reply:
It think its ok for now. I prefer to keep the PR lighter this way and implement the guard in further PR